### ISSUE TITLE
Filter by deployment status

### ISSUE THREAD

**Issue by Sandros94** [nuxt-hub/feedback#6]
**Is your feature request related to a problem? Please describe.**
By default Cloudflare Page's builds are triggered on each commit/PR to the configured production branch. This means that the `Deployments` page on Nuxt Hub will show a list `>=` commits/PRs to that branch.

Although the majority of times we could be interested only in the latest builds, it could be useful to be able to filter based on a particular deployment status, for long term debugging (Checking only `canceled` or `idle` or `failure`).

**Describe the solution you'd like**
The addition of a [SelectMenu](https://ui.nuxt.com/components/select-menu) to be able to pick what statuses to show or not, will all statuses selected by default (since @Atinux has already pointed out that this is would be only client side).

**Additional context**
This idea came to mind when setting up a test project, that I know I will handle it mostly in a single branch, with a good number of commits just to test features here and there, that wouldn't require a build. To do so I've disabled the automatic builds from the Cloudflare Page Settings for that particular project, and configured a Github action that will trigger a [Cloudflare Deploy Hook](https://developers.cloudflare.com/pages/configuration/deploy-hooks/) whenever I decide to.
But now there will be a lot of `idle` builds, since Cloudflare will detect those commits, but not fire a build.
![image](https://github.com/nuxt-hub/feedback/assets/13056429/a59ee61a-cf05-4b9f-bc0e-5e28d9a60c9b)

**Describe alternatives you've considered**
Change the `idle` badge color to gray to match the `canceled` one, or more simply make it different from the `failure` color (since, to my understanding, a deployment is tagged as `idle` when `ready but not fired`).

**Comment by Atinux** [comment]
We changed the `idle` badge color to `gray` so far.

We are still thinking about the ability to filter on client-side but the pagination might be very tricky to keep a good UX.

**Comment by Sandros94** [comment]
Agree. Lets see if it is something they could add, otherwise it would be simpler to leave as is, since I don't imagine this could get much attention

**Comment by dario-piotrowicz** [comment]
Hi @Sandros94 👋

I work at Cloudflare, @Atinux contacted us regarding this and it should be fairly easy to implement the filtering on our side, I'll take a look into this soon 🙂 :+1:

**Comment by dario-piotrowicz** [comment]
Hey @Sandros94, just to keep you updated, I did have a look at our internals and this unfortunately does not really look as easy as I initially thought 😓 

it would require a bunch of refactoring and the one described here is the only use case we have for this

I don't have too much capacity to spend much time on it so it might take a bit... or it could actually turn out complex enough not to make too much sense for us to implement in on our end (if there's no more demand for it)...

so it's a bit up in the air right now, but I will still try to see if I can help here 😓 🙏
